File: api/src/main/java/net/blueshell/api/repository/GuestRepository.java
package net.blueshell.api.repository;

import net.blueshell.api.base.BaseRepository;
import net.blueshell.api.model.Guest;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface GuestRepository extends BaseRepository<Guest, Long> {

    Optional<Guest> findByAccessToken(String accessToken);
}

File: api/src/main/kotlin/net/blueshell/api/repository/GuestRepository.kt
package net.blueshell.api.repository

import net.blueshell.api.model.event.Guest
import net.blueshell.api.repository.base.BaseRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface GuestRepository : BaseRepository<Guest, Long> {
    fun findByAccessToken(accessToken: String): Optional<Guest>
}
